### ISSUE TITLE
[0.7.0] Deprecate `get_relations_by_prefix` in favor of `get_relations_by_pattern`

### ISSUE THREAD

**Issue by clrcrl** [dbt-labs/dbt-utils#269]
### Describe the feature
I know, we are _just_ about to fully remove `get_tables_by_prefix` in favor of `get_relations_by_prefix`, and here I go saying "now let's remove `get_relations_by_prefix`!"

In the time it took us to do the first deprecation, we introduced `get_relations_by_pattern`, which, IMO, is a much nicer pattern. These two are functionally the same:
```
{% set relations = dbt_utils.get_relations_by_prefix(
    schema='my_schema',
    prefix='my_table_prefix'
) %}

{% set relations = dbt_utils.get_relations_by_pattern(
    schema_pattern='my_schema',
    table_pattern='my_table_prefix%'
) %} 

```
Note that `prefix` arg is now a `pattern`, so the value requires a trailing `%`.

(See [this comment](https://github.com/fishtown-analytics/dbt-utils/pull/220#issuecomment-644260242)).

Just to cut bloat down, I think we should:
- v0.7.0: mark the `get_relations_by_prefix` macro as deprecated, with migration instructions for `get_relations_by_pattern`. Also the `get_tables_by_prefix_sql` macro (in favor of `get_tables_by_pattern_sql`).
- v0.8.0 (or later): remove the `get_relations_by_prefix` and `get_tables_by_prefix_sql` macros

### Describe alternatives you've considered
Do nothing :) 

Also, if we ever do [this issue](https://github.com/fishtown-analytics/dbt/issues/1212) these macros become far less useful anyway! So the investment in maintaining them might not be worth it.

### Additional considerations

**`database_pattern` instead of `database`?**
I also considered whether we want to adapt `get_relations_by_pattern` to have a `database_pattern` arg. This would be nice for consistency, but would mean a longer deprecation path.

Given `information_schema.tables` is usually scoped to a database, it would probably look something like querying for the available databases, and then looping through the results.
- Snowflake: `show databases like '{{ database_pattern }}'`
- BigQuery: I don't think there's a SQL way to do this.
- Redshift: lol jk, you only have access to one database at a time anyway

Overall, that sounds too hard, and for little benefit, so I'd prefer to not do it 😬 

**Reassessing other arg names**
Basically, before we do this, we should take a moment to consider whether the arguments on this macro are right:
- Should we rename `exclude` to `exclude_table_pattern`? `exclude_identifier_pattern`?
- Should we add an additional `exclude_schema_pattern` argument?
- Should each argument be a list such that you can use multiple patterns? (IMO, no, just use the macro twice lol)
- Should we rename the `table_pattern` arg to `identifier_pattern` instead?

### Who will this benefit?
Very much maintainers only as we'll be able to remove some code. That being said, it also means we have to do the work to deprecate the macros.

### Are you interested in contributing this feature?
Yeah this one would be on me
